src/platform/EFR32/OTAImageProcessorImpl.cpp

```diff
@@ -174,18 +204,29 @@ void OTAImageProcessorImpl::HandleProcessBlock(intptr_t context)
         return;
     }
 
-    err = bootloader_eraseWriteStorage(mSlotId, mWriteOffset, (uint8_t *) (block.data()), block.size());
-
-    if (err)
+    // Copy data into the word-aligned writeBuffer and once it fills write its contents to the bootloader storage
+    uint32_t blockReadOffset = 0;
+    while (blockReadOffset < block.size())
```

Comment:
I can't use memcpy precisely because the block is not guaranteed to be a multiple of 64. I copy as much as I can into the write buffer and then wait for the next chunk of the image to be received OTA -- only then I will write copy the write buffer into the bootloader storage. The padding for the last chunk is handled in  HandleFinalize()

Reply:
Please add a comment saying that final data block (and any error checking) is done in HandleFinalize.

Reply:
Done